R/prepare_historical_xls.R

```diff
@@ -0,0 +1,176 @@
+
+#' Takes a file directory that contains
+#' Folders of historical .xls
+#' and processes them all into one df
+#' @importFrom purrr map_dfr
+#' @param directory the mbbs directory that has folders containing historical xls files. 
+hist_xls_purrr_directories <- function(directory = "inst/extdata/stop_level_data"){
+
+  filenames <- list.files(directory)  #get list of folders in directory
+  filenames <- filenames[str_detect(filenames,"\\_stops$")] #only keep folders ending in _stops
+  filenames <- paste(directory,filenames,sep="/") #add in the rest of the file path
+  stopsmbbs <- purrr::map_dfr(filenames, hist_xls_purrr_files_in_directory)
+  
+  return(stopsmbbs)
+}
+
+#' Takes a file directory of historical .xls
+#' files, and processes them,
+#' returning a dataframe that contains
+#' all the information from that directory's files
+#' broken down to the stop level.
+#' @importFrom purrr map_dfr
+#' @importFrom stringr str_detect
+#' @importFrom tidyr pivot_longer
+#' @importFrom dplyr %>%
+#' @param directory a directory that has folders containing historical xls files. Has a default for testing purposes
+hist_xls_purrr_files_in_directory <- function(directory = "inst/extdata/stop_level_data/Pippen_Durham_stops") {
+  
+  filenames <- list.files(directory) #list directory files
+  filenames <- filenames[str_detect(filenames, "\\.xls$")] #keep only .xls files
+  filenames <- paste(directory, filenames, sep = "/") #add in the rest of the path
+  stopsmbbs <- purrr::map_dfr(filenames, hist_xls_process_xls) %>% #process all xls in the directory
```

Comment:
Does this function return anything?

Reply:
stopsmbbs

Reply:
Does it?  For example, this function doesn't return anything:

```r
f <- function(x) {
  y <- x + 1
}
```

I think you don't want to assign the final expression to `stopmbbs`.